### PR TITLE
dyno: misc. scope resolve fixes

### PR DESCRIPTION
**Pull request by benharsh**
This PR includes three distinct changes to ``convert-uast`` that resolve some failures when scope-resolving the primers with ``--dyno``.

1) Adds new cases to ``convertScanReduceOp`` for word-based scan/reduce operations (e.g. minmax)
2) Does not scope-resolve function calls - this needs to be handled by the production compiler's type resolution process
3) Invokes ``noteConvertedSym`` when processing a ``TupleDecl``.

This PR also makes a small change to ``convertCalledKeyword``. Previously this function was resolving *any* identifier, rather than just keywords, and could result in some confusing situations. Removing this functionality does not result in any additional failures for the primers.

Testing:
- [x] full local (linux64)
- [x] primers --dyno (two failures remaining)

### CODE REVIEW

File: compiler/passes/convert-uast.cpp [hunk 1]
```diff
@@ -1140,6 +1140,17 @@ struct Converter {
         return new UnresolvedSymExpr("BitwiseOrReduceScanOp");
       if (name == USTR("^"))
         return new UnresolvedSymExpr("BitwiseXorReduceScanOp");
+
+      if (name.str() == "max")
```
Comment (mppf) [inline compiler/passes/convert-uast.cpp:1144]: Let's go ahead and use `USTR("max")` etc for these. You just have to add to compiler/dyno/include/chpl/framework/all-global-strings.h . I don't think we should be shy about adding these.

File: compiler/passes/convert-uast.cpp [hunk 2]
```diff
@@ -2027,7 +2037,16 @@ struct Converter {
       ret = isCallExpr(expr) ? toCallExpr(expr) : new CallExpr(expr);
       addArgsTo = ret;
     } else {
-      ret = new CallExpr(convertAST(calledExpression));
+      Expr* baseExpr = convertAST(calledExpression);
+      if (SymExpr* se = toSymExpr(baseExpr)) {
+        if (isFnSymbol(se->symbol())) {
+          ret = new CallExpr(new UnresolvedSymExpr(se->symbol()->name));
```
Comment (mppf) [inline compiler/passes/convert-uast.cpp:2043]: Can we put a comment saying this is a workaround? A reasonable alternative would be to adjust the Resolve.cpp code to bail out before establishing the type of a called function when there is just one candidate. Either way, I don't think we will want to generate UnresolvedSymExpr here in the long term, when we are using the dyno resolver to fully resolve calls.

**Review by mppf — APPROVED** [review]

File: compiler/passes/convert-uast.cpp (hunk 1 above)
Comment (benharsh) [reply compiler/passes/convert-uast.cpp]: Done 👍

File: compiler/passes/convert-uast.cpp (hunk 2 above)
Comment (benharsh) [reply compiler/passes/convert-uast.cpp]: I just pushed a commit to adjust Resolver.cpp, which fixes the ``sparse.chpl`` failure. I also added a condition to the parenless-function case for good measure.

**Pull request merged**